docs/serial_driver.md

```diff
@@ -1,129 +1,256 @@
 # 'serial' Driver
-This driver powers the [Split Keyboard](feature_split_keyboard.md) feature.
+
+The serial driver powers the [Split Keyboard](feature_split_keyboard.md) feature. Several implementations are available, depending on the platform of your split keyboard. Note that none of the drivers support split keyboards with more then two halves.
+
+| Driver                                  | AVR                | ARM                | Connection between halves                                                                     |
+| --------------------------------------- | ------------------ | ------------------ | --------------------------------------------------------------------------------------------- |
+| [Bitbang](#bitbang)                     | :heavy_check_mark: | :heavy_check_mark: | Single wire communication. One wire is used for reception and transmission.                   |
+| [USART Half-duplex](#usart-half-duplex) |                    | :heavy_check_mark: | Efficient single wire communication. One wire is used for reception and transmission.         |
+| [USART Full-duplex](#usart-full-duplex) |                    | :heavy_check_mark: | Efficient two wire communication. Two distinct wires are used for reception and transmission. |
 
 ?> Serial in this context should be read as **sending information one bit at a time**, rather than implementing UART/USART/RS485/RS232 standards.
 
-Drivers in this category have the following characteristics:
-* bit bang and USART Half-duplex provide data and signaling over a single conductor
-* USART Full-duplex provide data and signaling over two conductors
-* They are all limited to single master and single slave communication scheme
+<hr>
+
+## Bitbang
+
+This is the Default driver, the absence of configuration assumes this driver. It works by [bit banging](https://en.wikipedia.org/wiki/Bit_banging) a GPIO pin using the CPU. It is therefore not as efficient as a dedicated hardware peripheral, which the Half-duplex and Full-duplex drivers use.
+
+!> On ARM platforms the bitbang driver causes connection issues when using it together with the bitbang WS2812 driver. The Half-duplex or Full-duplex driver should be used whenever possible instead.
 
-## Supported Driver Types
+### Pin configuration
 
-|                   | AVR                | ARM                |
-| ----------------- | ------------------ | ------------------ |
-| bit bang          | :heavy_check_mark: | :heavy_check_mark: |
-| USART Half-duplex |                    | :heavy_check_mark: |
-| USART Full-duplex |                    | :heavy_check_mark: |
+```
+  LEFT                      RIGHT
++-------+      SERIAL     +-------+
+|   SSP |-----------------| SSP   |
+|       |       VDD       |       |
+|       |-----------------|       |
+|       |       GND       |       |
+|       |-----------------|       |
++-------+                 +-------+
+```
```

Comment:
I think these diagrams may not be very clear to new users, maybe a picture using a popular controller would be a better option?

Reply:
Yes indeed, although I would like to tackle that in a separate PR if that is fine with youß